lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3336,8 +3336,16 @@ namespace {
         return property->getParsedAccessor(AccessorKind::Set);
       }
 
-      // Emit diagnostics for incorrect usage of "returns_unretained" and
-      // "returns_unretained" attributes
+      emitdiagnosticsForReturnsRetainedUnretainedAttrs(decl);
+
+      return importFunctionDecl(decl, importedName, correctSwiftName,
+                                std::nullopt);
+    }
+
+    /// Emit diagnostics for incorrect usage of SWIFT_RETURNS_RETAINED and
+    /// SWIFT_RETURNS_UNRETAINED
+    void emitdiagnosticsForReturnsRetainedUnretainedAttrs(
```

Comment:
Keeping it for code clarity.

Reply:
Could you elaborate on what you mean?

Reply:
I opted to create a separate function,` emitDiagnosticsForReturnsRetainedUnretainedAttrs`, rather than inlining the logic within VisitFunctionDecl, primarily to enhance code clarity and avoid unnecessarily complicating VisitFunctionDecl. I noticed that similar helper methods are used elsewhere in this file, which aligns with this approach. While I understand the concern that this helper function is currently called only once, I believe maintaining helper functions where they improve code readability and maintainability is a good practice.

Reply:
Alright. This function still needs to be renamed to upper-case `Diagnostics` properly, as John suggested.